### PR TITLE
Stabilize cluster inventory

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

File: pkg/cluster/statusfilter.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+func (sf *statusFilter) Filter(dbType db.Type, statusColHdr *db.ColumnHandler) (string, error) {
+	statusColName, err := statusColHdr.ColumnName("Status")
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s IN ('%s')", statusColName, strings.Join(sf.stausesToStrings(), "','")), nil
+}
+
+func (sf *statusFilter) stausesToStrings() []string {
```
Comment (rJankowski93) [inline pkg/cluster/statusfilter.go:28]:
```suggestion
func (sf *statusFilter) statusesToStrings() []string {
```

**Review by rJankowski93 — CHANGES_REQUESTED** [review]

File: cmd/service/start/cmd.go [hunk 2]
```diff
@@ -85,58 +138,85 @@ func createOrUpdate(o *Options, w http.ResponseWriter, r *http.Request) {
...
+	//respond status URL
+	response := response(clusterState)
+	response["statusUrl"] = fmt.Sprintf("%s%s/%s/configs/%d/status", r.Host, r.URL.RequestURI(), clusterState.Cluster.Cluster, clusterState.Configuration.Version)
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		sendError(w, http.StatusInternalServerError, errors.Wrap(err, "Failed to generate progress URL response"))
+		return
 	}
 }
 
 func get(o *Options, w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply cmd/service/start/cmd.go:168]: Right now, the only consumer of the REST-interface is KEB. Would propose to extend the interface as soon as other parties are also connecting their system with the reconciler service.

File: cmd/service/start/cmd.go [hunk 3]
```diff
@@ -85,58 +138,85 @@ func createOrUpdate(o *Options, w http.ResponseWriter, r *http.Request) {
...
 	clusterState, err := o.Inventory().Get(cluster, configVersion)
 	if err != nil {
-		sendError(w, errors.Wrap(err, "Cloud not retrieve cluster state"))
+		sendError(w, http.StatusInternalServerError, errors.Wrap(err, "Cloud not retrieve cluster state"))
+		return
 	}
-	if err := json.NewEncoder(w).Encode(clusterState.Status.Status); err != nil {
-		sendError(w, errors.Wrap(err, "Failed to encode cluster status response"))
+	//respond
+	if err := json.NewEncoder(w).Encode(response(clusterState)); err != nil {
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply cmd/service/start/cmd.go:186]: totally right.. that's a missing piece! Thanks for fixing it!

File: pkg/cluster/statusfilter.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+type statusSQLFilter interface {
+	Filter(dbType db.Type, colHdr *db.ColumnHandler) (string, error)
+}
+
+type statusFilter struct {
+	allowedStatuses []model.Status
+}
+
+func (sf *statusFilter) Filter(dbType db.Type, statusColHdr *db.ColumnHandler) (string, error) {
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply pkg/cluster/statusfilter.go:20]: That's coming from the interface definition (the interval-Filter below requires it).

**Review by rJankowski93 — APPROVED** [review]

**Pull request merged**